### PR TITLE
fix: support SHOW query types in jdbc client

### LINKED ISSUE #4796: DBeaver does not support DESCRIBE SELECT * FROM "path"
### What happens?

When running query `DESCRIBE SELECT * FROM "path"` there is no results (0 rows fetched).
It works fine with tables.

DBeaver Version 22.2.0.202209051344
JDBC driver with DuckDB downloaded via DBeaver dialog (i think it downloads from Maven?) version 0.5.1

### To Reproduce

Run `DESCRIBE SELECT * FROM "path"` on any file (tested with csv and parquet in an inmemory database).

### OS:

Linux and Windows

### DuckDB Version:

0.5.1

### DuckDB Client:

DBeaver JDBC driver

### Full Name:

Tomasz Taraś

### Affiliation:

DXC

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Mause**
Fixes #4796

### CODE REVIEW

File: tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java [hunk 1]
```diff
@@ -103,7 +103,12 @@ private void prepare(String sql) throws SQLException {
 			params = new Object[0];
 			// TODO add query type to meta
 			String query_type = DuckDBNative.duckdb_jdbc_prepare_type(stmt_ref);
-			is_update = !query_type.equals("SELECT") && !query_type.equals("PRAGMA") && !query_type.equals("EXPLAIN");
```
Comment (Mytherin) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java:106]: In C++ the query result has a [statement return type](https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/common/enums/statement_type.hpp#L51) that could be used to determine if the query returns a result. Perhaps a good idea to also use here?

Comment (Mytherin) [reply tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java:106]: One thing to note is that this does not signify the query is an update or not, but only what the query returns. For example, insert statements can also return result sets:

```sql
D create table integers(i integer);
D insert into integers values (42) returning *;
┌────┐
│ i  │
├────┤
│ 42 │
└────┘
```

Comment (Mause) [reply tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java:106]: @Mytherin good point, I'll switch it over to that

File: tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java [hunk 2]
```diff
@@ -155,7 +155,7 @@ public ResultSet executeQuery() throws SQLException {
 
 	@Override
 	public int executeUpdate() throws SQLException {
-		if (!is_update) {
+		if (!returnsChangedRows) {
```
Comment (Mytherin) [inline tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java:158]: This means we can't use `executeUpdate` with catalog statements (e.g. `CREATE TABLE` - they return nothing) or with `INSERT INTO RETURNING` statements (which return a result set). Is that what we want?

Comment (Mause) [reply tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java]: You're correct yes, turns out I missed a line in the specification - `executeUpdate` should accept a query that updates rows, or returns nothing

> Executes the given SQL statement, which may be an INSERT, UPDATE, or DELETE statement or an SQL statement that returns nothing, such as an SQL DDL statement.

See https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#executeUpdate(java.lang.String)

I've fixed this, and reverted the test changes

**Review by Mytherin — APPROVED** [review]
Thanks! LGTM

**Pull request merged**